### ISSUE TITLE
[Candidate List] Filtering with Sex=Other does not show scanners

### ISSUE THREAD

**Issue by nicolasbrossard** [aces/Loris#6693]
**Describe the bug**
When I search for candidates with Sex=Other, I did not see any results on the test VM. 

**To Reproduce**
Access the candidate list module and select 'Other' in the Sex drop-down list.

**What did you expect to happen?**
I expected all scanners to be listed in the search result table

**Comment by driusan** [comment]
Does the candidate list module show scanners at all?

**Comment by nicolasbrossard** [comment]
Yeah, it does. There's even an 'Entity Type' (Human, Scanner) drop-down in the filter.

**Comment by laemtl** [comment]
@nicolasbrossard No scanner has data for sex, so I wonder if the observed behavior was expected. Selecting Sex=other displays correctly the Human Entity Type.

**Comment by nicolasbrossard** [comment]
@laemtl Right. It just seems weird to me that if you unite the result set you get when you select `Sex=M`, `Sex=F` and `Sex=O` you do not get the same result that you would get if you did not specify any criteria, since the scanners are listed in that case. But maybe that's not a big deal...I do not know what to do with this...

**Comment by laemtl** [comment]
@driusan @christinerogers Do you have inputs on this?

**Comment by nicolasbrossard** [comment]
This still a bug on 2024SEP17 and requires a discussion IMHO. Maybe introduce a new choice in the select drop-down, like `Not Applicable(Scanner)`?

**Comment by adamdaudrich** [comment]
for `scanner` :`female`, `male`, other, I get zero records
for `human` : `female`, `male` I get records, but for `other` I get no records

So @nicolasbrossard 
"if you unite the result set you get when you select Sex=M, Sex=F and Sex=O you do not get the same result that you would get if you did not specify any criteria, since the scanners are listed in that case" + "Maybe introduce a new choice in the select drop-down, like Not Applicable(Scanner)?

Just so I 'get' it - you want to show `Not Applicable` when `Scanner` is selected?
